Namespace(server_address='192.168.0.17:8080', rounds=1, sample_fraction=1.0, min_num_clients=4)
The IP address of the server is: 192.168.0.17
The IP address of the client is: 192.168.0.14, 192.168.0.22, 192.168.0.15, 192.168.0.19
The ID of the client is: pi

[rpi3b]
pi@raspberrypi3:~/E2FL$ python3.10 client_dataset.py --server_address=192.168.0.17:8080 --cid=0
/home/pi/.local/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
  warn(f"Failed to load image Python extension: {e}")
Namespace(server_address='192.168.0.17:8080', cid=0, dataset='cifar10', Model='ResNet20')
2024-06-20 14:54:44,227 - INFO - ['Wi-Fi start: 1718862884.2255528']
INFO flwr 2024-06-20 14:54:45,097 | grpc.py:52 | Opened insecure gRPC connection (no certificates were passed)
DEBUG flwr 2024-06-20 14:54:45,109 | connection.py:55 | ChannelConnectivity.IDLE
DEBUG flwr 2024-06-20 14:54:45,129 | connection.py:55 | ChannelConnectivity.READY
Client sampled for fit()
100%|███████████████████████████████████████████| 57/57 [05:03<00:00,  5.32s/it]
100%|███████████████████████████████████████████| 57/57 [05:01<00:00,  5.29s/it]
100%|███████████████████████████████████████████| 57/57 [05:00<00:00,  5.27s/it]
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:08<00:00,  4.45s/it]
DEBUG flwr 2024-06-20 15:10:29,279 | connection.py:220 | gRPC channel closed
INFO flwr 2024-06-20 15:10:29,281 | app.py:398 | Disconnect and shut down
2024-06-20 15:10:29,316 - INFO - ['Wi-Fi end: 1718863829.3144784']
2024-06-20 15:10:29,325 - INFO - The measurement data is saved as data_0_2024-06-20_15-10-29.pickle.

[rpi3b_2]
(env) pi@raspberrypi:~/E2FL $ python3.10 client_dataset.py --server_address=192.168.0.17:8080 --cid=1
/home/pi/.local/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
  warn(f"Failed to load image Python extension: {e}")
Namespace(server_address='192.168.0.17:8080', cid=1, dataset='cifar10', Model='ResNet20')
2024-06-20 14:54:34,679 - INFO - ['Wi-Fi start: 1718862874.6770432']
INFO :
INFO :      Received: train message fcdddb50-aa09-4858-b03f-04a85fe1c33b
Client sampled for fit()
100%|███████████████████████████████████████████| 57/57 [05:00<00:00,  5.27s/it]
100%|███████████████████████████████████████████| 57/57 [04:59<00:00,  5.25s/it]
100%|███████████████████████████████████████████| 57/57 [05:00<00:00,  5.27s/it]
INFO :      Sent reply
INFO :
INFO :      Received: evaluate message 7e70f496-d923-4c1f-8ad3-8a35063e6290
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:08<00:00,  4.43s/it]
INFO :      Sent reply
INFO :
INFO :      Received: reconnect message d7ce3c7d-3882-460d-84c3-8cf78550f9fc
INFO :      Disconnect and shut down
2024-06-20 15:10:29,146 - INFO - ['Wi-Fi end: 1718863829.144271']
2024-06-20 15:10:29,158 - INFO - The measurement data is saved as data_1_2024-06-20_15-10-29.pickle.

[rpi4]
pi@raspberrypi4:~/E2FL$ python3.10 client_dataset.py --server_address=192.168.0.17:8080 --cid=2
/home/pi/.local/lib/python3.10/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension:
  warn(f"Failed to load image Python extension: {e}")
Namespace(server_address='192.168.0.17:8080', cid=2, dataset='cifar10', Model='ResNet20')
2024-06-20 14:54:14,268 - INFO - ['Wi-Fi start: 1718862854.267578']
INFO flwr 2024-06-20 14:54:14,431 | grpc.py:52 | Opened insecure gRPC connection (no certificates were passed)
DEBUG flwr 2024-06-20 14:54:14,439 | connection.py:55 | ChannelConnectivity.IDLE
DEBUG flwr 2024-06-20 14:54:14,442 | connection.py:55 | ChannelConnectivity.READY
Client sampled for fit()
100%|███████████████████████████████████████████| 57/57 [01:23<00:00,  1.47s/it]
100%|███████████████████████████████████████████| 57/57 [01:27<00:00,  1.54s/it]
100%|███████████████████████████████████████████| 57/57 [01:36<00:00,  1.70s/it]
Client sampled for evaluate()
100%|█████████████████████████████████████████████| 2/2 [00:02<00:00,  1.29s/it]
DEBUG flwr 2024-06-20 15:10:29,281 | connection.py:220 | gRPC channel closed
INFO flwr 2024-06-20 15:10:29,281 | app.py:398 | Disconnect and shut down
2024-06-20 15:10:29,294 - INFO - ['Wi-Fi end: 1718863829.294012']
2024-06-20 15:10:29,296 - INFO - The measurement data is saved as data_2_2024-06-20_15-10-29.pickle.
